source/howto/AddUsersOpenStackVMInstance.rst

```diff
@@ -0,0 +1,148 @@
+##################################################
+Create additional user(s) on OpenStack VM/Instance
+##################################################
+
+Introduction
+############
+
+This guide explains how to create users in Linux using the command line and the *useradd* command and  grant administrator privileges using *sudo*.
+
+The `sudo` command provides a mechanism for granting administrator privileges — ordinarily only available to the root user — to normal users. 
+
+This guide will show you how to create a new user with `sudo` access on Linux, without having to modify your server’s `/etc/sudoers` file.
+
+Prerequisites
+#############
+
+* Valid Login and SSH or Console Access to VM
+* Have sudo privileges
+* VPN (if outside the network)
+
+Adding Users
+############
+
+#. Logging into you server
+
+    SSH in to your server as using your FEDADMIN user:
+
+
+    .. code-block:: shell
+
+      %ssh FEDID@172.16.113.47
+        ____ _____ _____ ____    ____ _                 _
+      / ___|_   _|  ___/ ___|  / ___| | ___  _   _  __| |
+      \___ \ | | | |_ | |     | |   | |/ _ \| | | |/ _` |
+        ___) || | |  _|| |___  | |___| | (_) | |_| | (_| |
+      |____/ |_| |_|   \____|  \____|_|\___/ \__,_|\__,_|
+
+                Location: r89.harwell.europe hpd r89rack241
+                  Branch: xfu59478/mr_build_centos_aq_image (sandbox)
+              Archetype: cloud
+            Personality: nubesvms
+        Operating System: centos7x-x86_64
+          Snapshot Date: 2019-05-20
+
+#. Login as Root
+
+    Use the `sudo` command to login in as root:
+
+    .. code-block:: shell
+
+      [FEDID@preprod-horizon1 ~]$ sudo su -
+
+      [root@preprod-horizon1 ~]#
+
+#. Adding a New User to the System using the script (addusers.sh) below. Edit the script and save it.
+
+    .. code-block:: shell
+
+      declare -A KEYS
+
+      KEYS[ntint]="ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDOXk3OukWtXQHbyhxuG2ST/zhGfoPHQi/5Zvt5PELskztdPMZRKn3bCqSk0nZQppoFgGsze93CWcfiY2ppUTGoKbZMnmDSjfO0DUFXB/ADNJapo53Yd0lRnxM45JuFT+AS8cg/TalzbBeXcO2Tyr/hFTm2p7b5PQZgny9RwdXI6W0rTyxnktBeMoy6bMepPueZgUhJVJIKFaAymvVlmeZHhlRX2MB/Y7OTpETNiuTbVaSw3kI57rZ16pgvCWwl0RyQon9V6yBsduaF9MAHTVlsuKBd7S7Q1V/19CpaAKdowq+HrZ5lbu3H7+GvX8LSpNAGzOtuNOxm50U/jcOWaUHxMtNCTMm1oHtdN13vADtTpU5MbtzJ89q/pSTfJVbX0oj8AFN+gtb4hlvKWaWwEUwOe5P4JlVIX4PjHZKzAdf+s1cumJ9sbjgvFIKzrGes6lpUBdjw0WGQnB0GLSkml7Hu6x8jecd4hC04tSqgJLw+KXZysVYCUkIuDEp/GKGv8aM= ntint@byod-vpn-168-0067.rl.ac.uk"
```

Comment:
I'm nervous about the fact you have you public keys in this doc page, a user blindly running the script will unintentionally add your public keys into their images. This opens us up to massive problems if the private key every becomes compromised or needs regenerating.

Ideally this should have instructions like:

- Copy the following file and add your keys with the username associated. 
- For example to create `example` with public key `AAAAB3NzaC1yc2EAAAAD` it would look like

```
.. code-block:: bash
  KEYS[example]="ssh-rsa AAAAB3NzaC1yc2EAAAAD..."
  groupadd sshusers
  for FEDID in "${!KEYS[@]}"
#      ...etc.
```

```